### PR TITLE
Convert Qt Enums to PyQt6 compatible forms

### LINKED ISSUE #1798: Add Enum class names to Enums for Qt
PyQt6 is using Python Enums for Qt Enum types, which means that you need to insert the Enum class name when referring to the Enum. See https://stackoverflow.com/questions/66235661/how-to-check-mousebuttonpress-event-in-pyqt6 for example.

So this means something like QEvent.Close becomes QEvent.Type.Close. This is a change with huge impacts in many places , but the construct is compatible with other Qt versions that we care about (PySide2, PySide6, PyQt5).

See https://github.com/enthought/pyface/issues/1092 for the Pyface version

### PR DESCRIPTION
**Pull request by corranwebster**
This should still work with all other Qt backends.  Fixes #1798.

One manual change was needed for the VideoEditor code. There is a change of enum names between Qt5 and Qt6 for QMediaPlayer, and I fixed to use the Qt5 names.  We will sort out how to handle the different names when we properly add support for Qt6 (which will likely require a number of other similar changes). 

**Checklist**
- [x] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

File: traitsui/qt4/table_editor.py [hunk 1]
```diff
@@ -1246,7 +1246,7 @@ def _update_header_sizing(self):
             hheader.setStretchLastSection(True)
         else:
             # hheader.setSectionResizeMode(
-            #     stretchable_columns[-1], QtGui.QHeaderView.Stretch)
+            #     stretchable_columns[-1], QtGui.QHeaderView.ResizeMode.Stretch)
```
Comment (aaronayres35) [inline traitsui/qt4/table_editor.py:1249]: just pointing out the commented code

File: traitsui/qt4/tabular_editor.py [hunk 2]
```diff
@@ -670,9 +670,9 @@ def paint(self, painter, option, index):
 
         # FIXME: 'styleHint' is returning bogus (negative) values. Why?
         # style = QtGui.QApplication.instance().style()
-        # color = style.styleHint(QtGui.QStyle.SH_Table_GridLineColor, option)
+        # color = style.styleHint(QtGui.QStyle.StyleHint.SH_Table_GridLineColor, option)
         # painter.setPen(QtGui.QColor(color))
```
Comment (aaronayres35) [inline traitsui/qt4/tabular_editor.py:674]: same here

File: traitsui/qt4/video_editor.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 # Thanks for using Enthought open source!
 
 """Traits UI 'display only' video editor."""
+from pyface.qt import is_qt5
```
Comment (aaronayres35) [inline traitsui/qt4/video_editor.py:12]: Doesn't seem like this is being used yet?

**Review by aaronayres35 — APPROVED** [review]
Changes LGTM

Do you have a reference for how you went about making the automated changes? A combo of PyQtEnumConverter and re.sub ?

**Comment by corranwebster** [comment]
> Do you have a reference for how you went about making the automated changes? A combo of PyQtEnumConverter and re.sub ?

Yes, the same script that I used for Pyface.

File: traitsui/qt4/table_editor.py (hunk 1 above)
Comment (corranwebster) [reply traitsui/qt4/table_editor.py:1249]: OK with changes to commented out code, since that code may eventually be not commented out.

**Pull request merged**